### PR TITLE
Maintenance week 07 2024

### PR DESCRIPTION
**Pull request by ehanson8**
### Purpose and background context
Updates app according to our [maintenance week documentation](https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/3445948428/DataEng+Application+Maintenance).

### How can a reviewer manually see the effects of these changes?
Run `make test` and `make` lint to confirm they still pass

### Includes new or updated dependencies?
YES

### Changes expectations for external applications?
NO

### What are the relevant tickets?
NA

### Developer
- [x] All new ENV is documented in README
- [x] All new ENV has been added to staging and production environments
- [x] All related Jira tickets are linked in commit message(s)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed and verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: carbon/app.py [hunk 1]
```diff
@@ -38,7 +38,9 @@ class CarbonFtpsTls(FTP_TLS):
         See ftplib.FTP_TLS for more details.
     """
 
-    def ntransfercmd(self, cmd: str, rest: str | int | None = None) -> tuple[socket, int]:
+    def ntransfercmd(
+        self, cmd: str, rest: str | int | None = None
+    ) -> tuple[socket, int | None]:
```
Comment (ehanson8) [inline carbon/app.py:43]: I was confused by this at first but it looks like the type hint was indeed updated, see #4 [here](https://github.com/python/typeshed/issues/10466)

Comment (ghukill) [reply carbon/app.py:43]: Nice find!  Helpful to understand why it changed.

File: carbon/feed.py [hunk 2]
```diff
@@ -70,11 +70,11 @@ def _add_element(self, record: dict[str, Any]) -> None | ET._Element:  # noqa: S
 
     def _add_subelement(
         self,
-        parent: ET._Element,  # noqa: SLF001
+        parent: ET._Element,
```
Comment (ghukill) [inline carbon/feed.py:73]: Minor, but always appreciate when we can remove linting overrides / ignores.

**Review by ghukill — APPROVED** [review]
Approved!  Thanks for the note about type hints changing.  Looks great all around, and agreed that the maintenance work is starting to establish a nice foundation.

**Review by jonavellecuerdo — CHANGES_REQUESTED** [review]
Code updates look good to me! 😄 One small change request is for the README. Can you update the environment variables section to the format:

---
## Environment Variables

### Required 

```bash
<ENV_VAR>=# <Description>.
<ENV_VAR>=# <Description>.
```

### Optional 

```shell
<ENV_VAR>=# <Description>.
<ENV_VAR>=# <Description>.
```

**Review by jonavellecuerdo — APPROVED** [review]

**Pull request merged**